tests/python/unittest/test_slice_tir.py

```diff
@@ -0,0 +1,178 @@
+#     the individual unit tests.
+#
+# Some of the unit tests below which require whole-IRModule comparison.  For expedience
+# we simply comment out the (early draft) bodies of those unit tests, rather than
+# hacking their structure to get the benefits of (a).
+
+
+# step 1: that vvvv simple passes Python / TVMScript parsing.
+#
+#   The only requirement for this test is that the TVMScript parser
+#   doesn't raise an error when encountering `T.call_tir(foo)`,
+#   where "foo" is a syntactically valid TVMScript function name.
+#
+#   NOTE! The role of this unit test should evolve as follows:
+#   1) Initially the test should fail, because we haven't yet changed the TVMScript
+#      parser to support 'call_tir'.
+#
+#   2) Initial TVMScript support for 'call_tir' will be minimal, essentially ignoring
+#      it.  This test should pass once that change is made.
+#
+#   3) As support for 'call_tir' becomes more complete, this test should once again
+#      fail, because the specified callee doesn't exist.  This test should be updated
+#      to once again expect failure.
+@pytest.mark.skip(reason="Awaiting TVMScript support for 'call_tir' token.")
+class TestParseCallTIR(tvm.testing.CompareBeforeAfter):
+    """
+    Simply confirm that the TIR node `call_tir` doesn't interfere with
+    the successful parsing of the TVMScript.
+    """
+
+    def before():
+        T.call_tir(add_one)
+        T.evalute(0)
+
+    def expected():
+        T.evaluate(0)
+
+
+# Step 2: transform annotated block ==> separate primfuncs + call_tir
+@pytest.mark.skip(
+    reason="Awaiting TVMScript support for 'call_tir' and T.annotation(\"functionalize\")."
+)
+class TestAnnotateAndSliceTIR(tvm.testing.CompareBeforeAfter):
+    # def test_annotate_and_slice():
+    #    @tvm.script.ir_module
+    #    class irmod_before:
+    #        @T.prim_func
+    #        def main(A: T.Buffer[(1,), "int8"):
+    #            #A = T.match_buffer(a, (1,), "int8")
+    #            A[0] = 0
+    #            with T.block("block_foo"): # optional: give this block a name, perhaps for testing?
+    #                # NOTE: nice to have: human control over name used for the generated callee
+    #                T.annotate("functionalize")  # TODO: find the actually correct name (and/or update the name in Eric's commit)
+    #                A[0] += 1
+    #                return 42
+    #
+    #    @tvm.script.ir_module
+    #    class irmod_after:
+    #        @T.prim_func
+    #        def main():
+    #            A = T.buffer[[1], "int8"]
+    #            A[0] = 0
+    #            with T.block():
+    #                call_tir(add_one, A)
+    #
+    #        # NOTE: it's not entirely clear how the name for the generated callee is chosen
+    #        @T.prim_func
+    #        def add_one(X: T.buffer[[1], "int8"]):
+    #            X[0] += 1
+    #
+    #    def create_schedule(self, irmod: tvm.ir.module.IRModule, blockname_to_funcname_map:Map[str,str]) -> tvm.tir.Schedule:
```

Comment:
What is the `create_schedule` function intended to do?

Reply:
I was (perhaps mistakenly) thinking that that slicing would involve the application of scheduling primitives.  I've removed this pseudocode for now, since that's a premature assumption.